app/src/androidTest/java/com/sdp/movemeet/view/navigation/security/ChatActivitySecurityTest.java

```diff
@@ -23,16 +27,20 @@
 
     @Before
     public void checkUserNotLogged() {
-       ChatActivity.enableNav = false;
         FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
         if (firebaseAuth.getCurrentUser() != null) firebaseAuth.signOut();
     }
 
-    //@Rule
-    //public ActivityScenarioRule<ChatActivity> testRule = new ActivityScenarioRule<>(ChatActivity.class);
-
     @Test
     public void redirectionTest() {
-        //onView(withId(R.id.button_login)).check(matches(isDisplayed()));
+        ChatActivity.enableNav = false;
```

Comment:
Could you remind me what is the purpose of setting `ChatActivity.enableNav` to false here? I'm actually not really sure to get the purpose of the `.enableNav` attribute...

Reply:
This technique was devised by Kepler. Each class has an enablenav attribute that allows to decide if the navigation menu should be instantiated or not. Indeed, for example in this case, the purpose of the test is to verify that a user who is not logged in and who arrives on the chat page is redirected to the login. In this case, when the menu is instantiated, the header will try to retrieve the information of a user that does not exist, leading to an error before the test even starts. By doing this, the menu will not be created so the navigation will not try to retrieve the data of a null user allowing the test to run properly

Reply:
Awesome, I see! Thank you for this thorough explanation!